test/template-tests/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1

```diff
@@ -0,0 +1,34 @@
+﻿param(
+[Parameter(Mandatory=$true,Position=0)]
+[PSObject]
+$MainTemplateObject
+)
+
+$allApiVersions = $MainTemplateObject | 
+    Find-AzureRMTemplate -Key apiVersion -Value * -Like
+
+foreach ($av in $allApiVersions) {
+    if ($av.ApiVersion -isnot [string]) {
+        Write-Error "Api Versions must be strings" -TargetObject $av -ErrorId ApiVersion.Not.String
+        continue
+    }
+    $helperName = if ($av.type) {
+        $av.type
+    } elseif ($av.name) {
+        $av.name
+    } else {
+        ''
+    }
+
+    $apiDate = $av.ApiVersion -as [DateTime]

```

Comment:
what if this contains "-preview"?

Reply:
Then the later cast to date will fail.  This effectively doesn't allow preview API versions, but provides a less than helpful error message in that scenario.  Did we want to allow preview API versions or not?

Reply:
we do want to allow preview apiVersions